addons/binding/org.openhab.binding.konnected/src/main/java/org/openhab/binding/konnected/handler/KonnectedHandler.java

```diff
@@ -0,0 +1,330 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.konnected.handler;
+
+import static org.openhab.binding.konnected.KonnectedBindingConstants.PIN_TO_ZONE;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.StateDescription;
+import org.openhab.binding.konnected.internal.KonnectedConfiguration;
+import org.openhab.binding.konnected.internal.KonnectedDynamicStateDescriptionProvider;
+import org.openhab.binding.konnected.internal.KonnectedHTTPUtils;
+import org.openhab.binding.konnected.internal.KonnectedPutSettingsTimer;
+import org.openhab.binding.konnected.internal.servelet.KonnectedHTTPServelet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link KonnectedHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Zachary Christiansen - Initial contribution
+ */
+
+public class KonnectedHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(KonnectedHandler.class);
+
+    private KonnectedConfiguration config;
+    private KonnectedHTTPServelet webHookServlet;
+    private KonnectedPutSettingsTimer putSettingsTimer;
+    private List<String> sensors;
+    private List<String> actuators;
+    private List<Boolean> isAct;
+    KonnectedHTTPUtils http;
+    private KonnectedDynamicStateDescriptionProvider dynamicStateDescriptionProvider;
+
+    public KonnectedHandler(Thing thing, KonnectedHTTPServelet webHookServlet,
+            KonnectedDynamicStateDescriptionProvider provider) {
+        super(thing);
+        this.webHookServlet = webHookServlet;
+        this.putSettingsTimer = new KonnectedPutSettingsTimer();
+        this.sensors = new LinkedList<String>();
+        this.actuators = new LinkedList<String>();
+        http = new KonnectedHTTPUtils();
+        this.isAct = new LinkedList<Boolean>();
```

Comment:
Please move the initialization of these variables to the properties above and make them final.
That helps against NPEs. You can `.clear()` these lists in initialize if you want to reset them.

Also an ArrayList has some performance benefits over a LinkedList that's why it is more commonly used:
See: https://dzone.com/articles/arraylist-vs-linkedlist-vs

Please also update the code to use the diamond operator where possible, e.g.:

`private List<String> sensors = new ArrayList<>();`

Reply:
i have switched the code over to using Gson so there are no lists now.